src/go/rpk/pkg/cli/cloud/cluster/secret/list.go

```diff
@@ -0,0 +1,67 @@
+package secret
+
+import (
+	v1alpha2 "buf.build/gen/go/redpandadata/dataplane/protocolbuffers/go/redpanda/api/dataplane/v1alpha2"
+	"connectrpc.com/connect"
+	"github.com/redpanda-data/redpanda/src/go/rpk/pkg/config"
+	"github.com/redpanda-data/redpanda/src/go/rpk/pkg/out"
+	"github.com/redpanda-data/redpanda/src/go/rpk/pkg/publicapi"
+	"github.com/spf13/afero"
+	"github.com/spf13/cobra"
+)
+
+func NewListCommand(fs afero.Fs, p *config.Params) *cobra.Command {
+	var nameContains string
+
+	cmd := &cobra.Command{
+		Use:   "list",
+		Short: "List all secrets",
+		Long:  "List all secrets in your Redpanda cloud cluster",
+		Run: func(cmd *cobra.Command, _ []string) {
+			// Load profile
+			p, err := p.LoadVirtualProfile(fs)
+			out.MaybeDie(err, "rpk unable to load config: %v", err)
+			if p.FromCloud {
+				makeRequest := func(url string) {
+					if url == "" {
+						out.Die("URL cannot be empty")
+					}
+					cl, err := publicapi.NewDataPlaneClientSet(url, p.CurrentAuth().AuthToken)
+					out.MaybeDie(err, "unable to initialize cloud client: %v", err)
+
+					// Create request to list secrets
+					request := &v1alpha2.ListSecretsRequest{
+						Filter: &v1alpha2.ListSecretsFilter{
+							NameContains: nameContains,
+						},
+					}
+					response, err := cl.Secrets.ListSecrets(cmd.Context(), &connect.Request[v1alpha2.ListSecretsRequest]{Msg: request})
+					out.MaybeDie(err, "unable to list secrets: %v", err)
+
+					// Print the list of secrets
+					tw := out.NewTable("name")
+					defer tw.Flush()
+					for _, secret := range response.Msg.Secrets {
+						tw.Line(secret.Id)
+					}
```

Comment:
This is not part of `makeRequest`, at least its not a part of the request but instead is printing the response.

Reply:
I remove all makeRequest fn